app/controllers/admin/publishers_controller.rb

```diff
@@ -30,30 +30,16 @@ def update
     redirect_to admin_publisher_path(current_publisher)
   end
 
-  # generates a publisher statement for an admin
-  # does not send an email
-  def generate_statement
-    publisher = Publisher.find(params[:id])
+  def statement
     statement_period = params[:statement_period]
-    statement = PublisherStatementGenerator.new(publisher: publisher,
-                                                statement_period: statement_period.to_sym,
-                                                created_by_admin: true).perform
-
-    SyncPublisherStatementJob.perform_later(publisher_statement_id: statement.id, send_email: false)
-    render(json: {
-      id: statement.id,
-      date: statement_period_date(statement.created_at),
-      period: statement_period_description(statement.period.to_sym)
-    }, status: 200)
-  end
-
-  def statement_ready
-    statement = PublisherStatement.find(params[:id])
-    if statement && statement.contents
-      head 204
-    else
-      head 404
-    end
+    @transactions = PublisherStatementGetter.new(publisher: @publisher, statement_period: statement_period).perform
+    @statement_period = publisher_statement_period(@transactions)
+    statement_file_name = "BravePaymentsStatemet-#{@statement_period}"
```

Comment:
Misspelling.

Also, is there a better way of handling users who have 0 transactions? I prefer displaying an error via `flash[:alert]` or `flash[:error]` if it's empty rather than accepting an empty string.

Reply:
I think it's best to just tell them they have no statements on the statements page and don't give them the option to download

Reply:
Though, if they have previous transactions in "all" but none from "this_month" then we'd still be giving them the option to download "this_month".  In that case it'd be good to flash an alert.

This is what it'd look like if they haven't ever had a transaction:
![image](https://user-images.githubusercontent.com/12549658/46027317-8bb34880-c0bb-11e8-9570-8a5077af6563.png)